packages/shorebird_cli/lib/src/commands/init_command.dart

```diff
@@ -210,7 +209,7 @@ Please make sure you are running "shorebird init" from within your Flutter proje
     );
 
     if (!shorebirdEnv.pubspecContainsShorebirdYaml) {
-      _addShorebirdYamlToPubspecAssets(
+      ShorebirdEnv.addShorebirdYamlToPubspecAssets(
```

Comment:
It doesn't need to be static – an instance method on shorebirdev would mean that we don't need to pass the pubspec file as an argument

Reply:
Yep! ShorebirdEnv is a dependency of both the InitCommand and ShorebirdYamlAssetValidator. For unit tests, we assume dependencies do what they say. That is, a unit test for the InitCommand should only be testing the code in the InitCommand, and can assume that, as long as `shorebirdEnv. addShorebirdYamlToPubspecAssets()` was called, it's doing the right thing. We would ensure that `shorebirdEnv. addShorebirdYamlToPubspecAssets()` does what it should in the ShorebirdEnv unit tests.

Reply:
I took a stab at it @bryanoltman 

Is something like this what you had in mind? 

Might still need one more ShorebirdEnv unit test with this approach, but I want to confirm the direction first